guides/pages/how-to-add-users-to-liveblocks-comments.mdx

```diff
@@ -0,0 +1,338 @@
+    width={1291}
+    height={892}
+  />
+</Figure>
+
+## What we’re learning
+
+In this guide we’ll be modifying
+[`createRoomContext`](/docs/api-reference/liveblocks-react#createRoomContext) in
+our config file, learning how to:
+
+- Add names and avatars to threads and comments with
+  [`resolveUsers`](/docs/api-reference/liveblocks-react#resolveUsers).
+- Create user mention suggestions in the composer using
+  [`resolveMentionSuggestions`](/docs/api-reference/liveblocks-react#resolveMentionSuggestions).
+
+## Authenticate your application
+
+The first step is to find an
+[authentication guide for your framework](/docs/rooms/authentication) and
+authenticate your app, as this is necessary for Comments.
+
+Make sure to follow the metadata step in the guide, and attach the name of your
+user, along with the URL of their avatar, as these properties will both be used
+in the [default components](/docs/products/comments#Components). Here’s an
+example using access token authentication, with an email address as a user’s ID.
+
+```ts title="Metadata in access tokens"
+const session = liveblocks.prepareSession("marc@example.com", {
+  userInfo: {
+    name: "Marc",
+    avatar: "https://example.com/marc.png",
+
+    // Your custom metadata
+    // ...
+  },
+});
+```
+
+If you’re using ID token authentication, it’ll look a little different.
+
+```ts title="Metadata in ID tokens" isCollapsed isCollapsable
+const { status, body } = await liveblocks.identifyUser(
+  {
+    userId: "marc@example.com",
+    groupIds: [], // Optional
+  },
+  {
+    userInfo: {
+      name: "Marc",
+      avatar: "https://example.com/marc.png",
+
+      // Your custom metadata
+      // ...
+    },
+  }
+);
+```
+
+Don’t forget to modify your `UserMetadata` type in `liveblocks.config.ts` to
+match the metadata format, adding type hints to your editor.
+
+```ts file="liveblocks.config.ts"
+type UserMetadata = {
+  id: string;
+
+  info: {
+    name: string;
+    avatar: string;
+
+    // Your custom metadata
+    // ...
+  };
+};
+```
+
+Any metadata you add is made saved and accessible in your Comments app with the
+[`useUser`](/docs/api-reference/liveblocks-react#useUser) hook.
+
+## Resolving users
+
+To show each user’s name and avatar in threads and comments, we need to use
+[`resolveUsers`](https://liveblocks.io/docs/api-reference/liveblocks-react#resolveUsers).
+
+<Figure>
+  <Image
+    src="/assets/tutorials/comments/thread.png"
+    alt="Thread with resolved users"
+    width={1291}
+    height={892}
+  />
+</Figure>
+
+<Steps>
+  <Step>
+    <StepTitle>Add the function to your config file</StepTitle>
+    <StepContent>
+      The `resolveUsers` function is passed as an option to
+      [`createRoomContext`](/docs/api-reference/liveblocks-react#createRoomContext)
+      at the bottom of your `liveblocks.config.ts` file—let’s add it. This function
+      provides you with `userIds`, an array of user IDs that have created a thread or comment.
```

Comment:
```suggestion
      provides you with `userIds`, an array of user IDs that are present in the current room (e.g. they have created a thread, a comment, added a reaction, etc.).
```

Reply:
I don't like `present` because that's obviously clashing with Presence but I think we don't want to convey a too strict idea of why user IDs would come up here, they're appear as soon a user is visible so that includes many things, even things like mention suggestions, etc.